docs/reference/modules/cluster/shards_allocation.asciidoc

```diff
@@ -98,13 +98,33 @@ Specify when shard rebalancing is allowed:
 --
 
 `cluster.routing.allocation.cluster_concurrent_rebalance`::
-      (<<dynamic-cluster-setting,Dynamic>>)
-      Allow to control how many concurrent shard rebalances are
-      allowed cluster wide. Defaults to `2`. Note that this setting
-      only controls the number of concurrent shard relocations due
-      to imbalances in the cluster. This setting does not limit shard
-      relocations due to <<cluster-shard-allocation-filtering,allocation
-      filtering>> or <<forced-awareness,forced awareness>>.
+(<<dynamic-cluster-setting,Dynamic>>)
+Defines the number of concurrent shard rebalances are allowed across the whole
+cluster. Defaults to `2`. Note that this setting only controls the number of
+concurrent shard relocations due to imbalances in the cluster. This setting does
+not limit shard relocations due to
+<<cluster-shard-allocation-filtering,allocation filtering>> or
+<<forced-awareness,forced awareness>>.
+
+`cluster.routing.allocation.type`::
++
+--
+Selects the algorithm used for computing the cluster balance. Defaults to
+`desired_balance` which selects the _desired balance allocator_. This allocator
+runs a background task which computes the desired balance of shards in the
+cluster by repeatedly simulating shard movements until no shard movement
+improves the balance of the cluster. Once this background task completes, {es}
+moves shards to their desired locations.
```

Comment:
This is obviously pretty brief but I'm not sure we need to go into much more detail here today. There's not much user-visible stuff in this area yet. I expect we'll extend this and perhaps break it out into its own section once we have more APIs and things.

Reply:
```
computes the desired balance of shards in the
cluster by repeatedly simulating shard movements until no shard movement
improves the balance of the cluster
```
this sounds a lot like implementation detail to me. Should we consider:
```
computes the desired balance of shards in the cluster until shard counts and their 
resource usages are as close to even distribution as allowed by allocation settings
```
?

Reply:
Agree, I suggested an even simpler version of it.